### ISSUE TITLE
Bug in average_precision Metric

### ISSUE THREAD

**Issue by elias-ramzi** [PyTorchLightning/pytorch-lightning#2315]
## 🐛 Bug

Hi everyone, I encountered a bug when using the average_precision metric (pytorch_lightning.metrics.functional.classification). It yields incorrect results (negative ones).

There seems to be a missing parenthesis in the code here : 

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/functional/classification.py#L847

It works when corrected as : 

`return -torch.sum((recall[1:] - recall[:-1]) * precision[:-1])`

In order to reproduce negative results :

```
import torch
import pytorch_lightning.metrics.functional.classification as M

torch.manual_seed(23)
truth = (torch.rand(100) > .6)
pred = torch.rand(100)

M.average_precision(pred, truth)
```

I did not find an issue on this topic yet. If needed I can submit a PR.

Thanks :relaxed:

**Comment by InCogNiTo124** [comment]
I'd like to fix it

**Comment by elias-ramzi** [comment]
Hi, I have already created a branch for the PR !

**Comment by edenlightning** [comment]
Thanks for the issue and the fix! @InCogNiTo124 Just saying- there are many other open issues if you want to take a stab at :)
